src/dbus_api/pool.rs

```diff
@@ -219,6 +219,61 @@ fn snapshot_filesystem(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
     Ok(vec![msg])
 }
 
+fn init_cache_2_1(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
```

Comment:
There seems to be so much code shared between ```add_blockdevs``` and this method that there is probably a lot of room for abstraction and unification between the two.

Reply:
Resolved.